pkg/collector/corechecks/cluster/kubernetes_apiserver.go

```diff
@@ -71,12 +78,63 @@ func (k *KubeASCheck) Run() error {
 
 	componentsStatus, err := asclient.ComponentStatuses()
 	if err != nil {
-		k.Warn("could not retrieve the status from the control plane's components", err.Error())
+		k.Warnf("Could not retrieve the status from the control plane's components %s", err.Error())
 	}
 
 	err = k.parseComponentStatus(sender, componentsStatus)
 	if err != nil {
-		k.Warn("could not collect API Server component status: ", err.Error())
+		k.Warnf("Could not collect API Server component status: %s", err.Error())
+	}
+
+	var token string
+
+	if k.latestEventToken == 0 {
```

Comment:
You can move the ConfigMapTokenFetcher call inside the `Config()` method, and don't need to check `k.latestEventToken == 0` at every run